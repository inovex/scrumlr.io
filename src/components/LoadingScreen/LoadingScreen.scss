@import "constants/style";

.loading-screen {
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

[theme="dark"] {
  .loading-screen {
    background-color: $color-dark-one;
  }
}
