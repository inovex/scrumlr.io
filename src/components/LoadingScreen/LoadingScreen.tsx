import "./LoadingScreen.scss";
import {LoadingIndicator} from "components/LoadingIndicator";

export const LoadingScreen = () => (
  <div className="loading-screen">
    <LoadingIndicator />
  </div>
);
