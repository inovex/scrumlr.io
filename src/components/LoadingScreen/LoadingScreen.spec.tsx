import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import LoadingScreen, {
  LoadingScreenProps
} from '../../components/LoadingScreen/LoadingScreen';

describe('<LoadingScreen />', () => {
  let wrapper: ShallowWrapper<LoadingScreenProps>;

  it('should render loading screen without status', () => {
    wrapper = shallow(<LoadingScreen />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render loading screen with status', () => {
    wrapper = shallow(<LoadingScreen status="Test status message" />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
