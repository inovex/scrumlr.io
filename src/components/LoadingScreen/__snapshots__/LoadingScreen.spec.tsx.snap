// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingScreen /> should render loading screen with status 1`] = `"<div class=\\"loading-screen\\"><h1 class=\\"loading-screen__headline\\">Scrumlr</h1><div class=\\"loading-indicator\\"><div class=\\"bounce1\\"></div><div class=\\"bounce2\\"></div><div class=\\"bounce3\\"></div></div><span class=\\"loading-screen__status\\">Test status message</span></div>"`;

exports[`<LoadingScreen /> should render loading screen without status 1`] = `"<div class=\\"loading-screen\\"><h1 class=\\"loading-screen__headline\\">Scrumlr</h1><div class=\\"loading-indicator\\"><div class=\\"bounce1\\"></div><div class=\\"bounce2\\"></div><div class=\\"bounce3\\"></div></div></div>"`;
